backend/event_processor/src/main/kotlin/com/feed_grabber/event_processor/fileStorage/AmazonS3ClientService.kt

```diff
@@ -20,15 +25,16 @@ class AmazonS3ClientService(private val s3Client: S3Client) {
     @Value("\${amazon.endpointUrl}")
     private lateinit var ENDPOINT: String
 
-    fun uploadReport(resource: Resource): String {
+    fun uploadReport(resource: File): String {
         return putObject(resource, "reports/")
     }
 
-    fun putObject(resource: Resource, prefix: String): String {
-        val key = prefix + LocalDateTime.now() + "-" + resource.filename?.replace(" ","_") ?: UUID.randomUUID().toString()
+    fun putObject(file: File, prefix: String): String {
+        val key = prefix + LocalDateTime.now() + "-" + file.name?.replace(" ","_") ?: UUID.randomUUID().toString()
         val request = PutObjectRequest.builder().bucket(BUCKET_NAME).key(key).acl(ObjectCannedACL.PUBLIC_READ).build()
-        val requestBody = RequestBody.fromInputStream(resource.inputStream, resource.contentLength())
-        s3Client.putObject(request, requestBody)
+        val multiFile: MultipartFile = MockMultipartFile("report.xlsx", "report.xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", file.readBytes())
```

Comment:
https://docs.aws.amazon.com/AmazonS3/latest/dev/UploadObjSingleOpJava.html maybe this is more suitable?